src/components/list/PagedList.js

```diff
@@ -0,0 +1,181 @@
+import Checkbox from '../ui/Checkbox'
+import LoadingIndicator from '../ui/LoadingIndicator'
+import SquareButton from '../ui/SquareButton'
+import EntryList from './EntryList'
+
+const LIMIT = 30
+
+const StyledContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  height: 100%;
+`
+
+const StyledListContainer = styled.div`
+  height: 100%;
+`
+
+const StyledPageInfo = styled.div`
+  display: flex;
+  flex-direction: column;
+  visibility: ${({ visible }) => (visible ? 'visible' : 'hidden')};
+`
+
+const StyledPageNav = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+  padding: 10px;
+  border-top: 1px solid ${({ theme }) => theme.secondaryBackground};
+`
+
+const StyledNavButtonContainer = styled.div`
+  display: flex;
+  button {
+    &:not(:last-child) {
+      margin-right: 4px;
+    }
+  }
+`
+
+const StyledCheckboxContainer = styled.div`
+  display: flex;
+  padding: 10px;
+  color: ${({ theme }) => theme.secondaryText};
+`
+
+const PagedList = () => {
+  const history = useHistory()
+  const container = useRef()
+  const rect = useRect(container) ?? { width: 0, height: 0 }
+  const { view } = useMap()
+  const [locations, setLocations] = useState([])
+  const [currentOffset, setCurrentOffset] = useState(0)
+  const [totalLocations, setTotalLocations] = useState(0)
+  const [updateOnMapMove, setUpdateOnMapMove] = useState(false)
+  const [currentView, setCurrentView] = useState(undefined)
+  const [loadingNextPage, setLoadingNextPage] = useState(false)
+
+  useEffect(() => {
+    const setInitialView = () => {
+      const { bounds, zoom } = view
+      if (
+        bounds?.ne.lat != null &&
+        zoom > 12 &&
+        (updateOnMapMove || currentView === undefined)
+      ) {
+        setCurrentView(view)
+        setCurrentOffset(0)
+        setLocations([])
+      } else if (zoom <= 12) {
+        setLocations([])
+        setCurrentView(undefined)
+        setCurrentOffset(0)
+      }
+    }
+    setInitialView()
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [view, updateOnMapMove])
+
+  useEffect(() => {
+    const fetchCurrentListEntries = async () => {
+      if (currentView !== undefined) {
+        setLoadingNextPage(true)
+        const { bounds, center } = currentView
+        const locations = await getLocations({
+          swlng: bounds.sw.lng,
+          nelng: bounds.ne.lng,
+          swlat: bounds.sw.lat,
+          nelat: bounds.ne.lat,
+          lng: center.lng,
+          lat: center.lat,
+          limit: LIMIT,
+          offset: currentOffset,
+        })
+        setTotalLocations(locations[1])
+        setLocations(locations.slice(2))
+        setLoadingNextPage(false)
+      }
+    }
+    fetchCurrentListEntries()
+  }, [currentOffset, currentView])
+
+  const handleCheckboxClick = () => setUpdateOnMapMove(!updateOnMapMove)
+
+  const handlePreviousPageClick = () => {
+    if (currentOffset > 0) {
```

Comment:
I don't think you need these if statements. Will disabled take care of not triggering onClick?

Reply:
If so, you can inline them inside `onClick={}`

Reply:
good catch